addons/portal/static/src/js/portal.js

```diff
@@ -282,6 +282,90 @@ publicWidget.registry.portalSecurity = publicWidget.Widget.extend({
 
 });
 
+/**
+ * Defining what happens when you click the "Log out from all devices" 
+ * on the "/my/security" page.
+ */
+publicWidget.registry.RevokeSessionsButton = publicWidget.Widget.extend({
+    selector: '#portal_revoke_all_sessions_popup',
+    events: {
+        click: '_onClick',
+    },
+
+    async _onClick(e) {
+        const rpc =this.proxy('_rpc')
+        const wrapped =this._rpc({
+            model: 'res.users',
+            method: 'api_key_wizard',
+            args: [session.user_id],
+        });
+        await handleRevokeSessions(rpc, wrapped, e);
```

Comment:
Seems a complicated JS code structure compared to what it does (a wrapper, a pointer, a function outside of widget scope, ...), unless I misunderstood something ?